### PR TITLE
NXP-27705: Add a bulk enrichment action

### PR DESCRIPTION
**Pull request by covolution**

### CODE REVIEW

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/BulkEnrichmentAction.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        protected void compute(CoreSession coreSession, List<String> ids, Map<String, Serializable> map) {
+            ModelServingService modelServingService = Framework.getService(ModelServingService.class);
+            for (String id : ids) {
+                try {
+                    DocumentModel doc = coreSession.getDocument(new IdRef(id));
+                    Set<String> inputs = modelServingService.getInputs(doc);
+                    if (!inputs.isEmpty()) {
+                        BlobTextFromDocument blobTextFromDocument = PropertyUtils.docSerialize(doc, inputs);
+                        if (blobTextFromDocument != null) {
+                            outputs.add(toRecord(blobTextFromDocument.getKey(), blobTextFromDocument));
```
Comment (andreinechaev) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/BulkEnrichmentAction.java]: how large outputs can be? Do we really need to keep the array on the class level?

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java [hunk 2]
```diff
@@ -132,7 +132,7 @@ protected void compute(CoreSession coreSession, List<String> ids, Map<String, Se
             for (String id : ids) {
                 try {
                     DocumentModel doc = coreSession.getDocument(new IdRef(id));
-                    BlobTextFromDocument subDoc = PropertyUtils.docSerialize(doc, customProperties);
+                    BlobTextFromDocument subDoc = PropertyUtils.docSerialize(doc, new HashSet<>(customProperties));
```
Comment (andreinechaev) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java]: why do you need to wrap it around? making it mutable?

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/BulkEnrichmentAction.java (hunk 1 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/BulkEnrichmentAction.java]: Yes, that's a bit nasty isn't it.  Its the way other batch computations.  You need the `ComputationContext` to write the record.  The batch size is set by config, so its 50.

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java (hunk 2 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/bulk/DataSetBulkAction.java]: I changed the type from List to Set

**Comment by andreinechaev** [comment]
👍

**Pull request merged**